index.js

```diff
@@ -0,0 +1,42 @@
+exports.backgroundColorChanger = function() {
+
+    // assign array to hold all hexadeximal integers
+    var hexArray = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 'a', 'b', 'c', 'd', 'e', 'f'];
+
+    function setColor() {
+        var colorArray = [];
+        // push random values from hexArray into colorArray
+        for (i = 0; i <= 5; i++) {
+            colorArray.push(hexArray[Math.floor(Math.random() * hexArray.length)]);
+        }
+        // join the colorArray values into a hexidecimal color code and return
+        var color = "#" + colorArray.join("");
+        console.log(color);
+        return color;
+    }
+
+    // change the background color every second and run 10 times
+    var timesRun = 0;
+    var usedColors = [];
+
+    function changeColor () {
+        var myColor = setColor();
+        // check for unlikely repeating colors
+        if (usedColors.indexOf(myColor) === -1 ) {
```

Comment:
@natation Since time complexity should be O(1) for accessing an object property, I can make this operation constant time by using an object rather than an array.  The values shouldn't matter, so I can set those to null and use hasOwnProperty on the keys to see if they're already added.

If we were going to cycle through thousands of colors, I'd also be concerned about the growing size of this object, but I think your question for now had to do with runtime performance.